src/App/menuUtils.ts

```diff
@@ -1,26 +1,23 @@
+import { Auth } from '../redux/mapStoreToProps';
 import commonUtils from '../lib/commonUtils';
+import { MenuItem } from './menuItems';
+import type { AppTemplate } from './AppTemplate';
 
-export interface MenuUtils {
-  menuItem: (...args: any) => JSX.Element | null
-}
-function continueMenuItem(menu:
-{ link: any; name?: string; type?: any; auth?: boolean }, index: any, location: { pathname: string | string[]; },
-auth: { isAuthenticated: any; },
-view: { props?: { location: any; auth: any; }; makeMenuLink?: any; googleButtons?: any; }): JSX.Element | null {
-  if (menu.link !== '') return view.makeMenuLink(menu, index);
-  if (menu.type === 'googleLogin' && !auth.isAuthenticated && location.pathname.includes('/staff')) return view.googleButtons('login', index);
+const continueMenuItem = (menu: MenuItem,
+  index: number, auth: Auth, view: AppTemplate): JSX.Element | null => {
+  if (menu.type === 'googleLogin' && !auth.isAuthenticated) return view.googleButtons('login', index);
   if (menu.type === 'googleLogout' && auth.isAuthenticated) return view.googleButtons('logout', index);
   return null;
-}
+};
 
-function menuItem(menu: { link: string; name?: string; type?: any; auth?: boolean },
-  index: any, view: { props: { location: any; auth: any; }; }): JSX.Element | null {
-  const userRoles: any[] = commonUtils.getUserRoles();
+function menuItem(menu: MenuItem,
+  index: number, view: AppTemplate): JSX.Element | null {
+  const userRoles: string[] = commonUtils.getUserRoles();
   const { location, auth } = view.props;
   if (location.pathname === '/staff' && menu.link === '/staff') return null;
   if ((menu.link === '/staff' || menu.link === '/belief') && auth.isAuthenticated) return null;
-  if (menu.name === 'Admin Dashboard' && (!auth.isAuthenticated || userRoles.indexOf(auth.user.userType) === -1)) return null;
-  return continueMenuItem(menu, index, location, auth, view);
+  if (menu.name === 'Admin Dashboard' && (!auth.isAuthenticated || userRoles.indexOf(auth.user.userType || '') === -1)) return null;
```

Comment:
@Unesdala the same edit here, so that the nav menu also does not display when the user has the wrong roles, thank you